raiden/routing.py

```diff
@@ -40,47 +49,124 @@ def get_best_routes(
     privkey: bytes,
 ) -> Tuple[Optional[str], List[RouteState], Optional[UUID]]:
 
-    is_direct_partner = to_address in views.all_neighbour_nodes(chain_state)
-    can_use_pfs = pfs_config is not None and one_to_n_address is not None
-
-    log.debug(
-        "Getting route for payment",
-        source=to_checksum_address(from_address),
-        target=to_checksum_address(to_address),
-        amount=amount,
-        target_is_direct_partner=is_direct_partner,
-        can_use_pfs=can_use_pfs,
-    )
-
-    # the pfs should not be requested when the target is linked via a direct channel
-    if is_direct_partner:
-        internal_routes = get_best_routes_internal(
-            chain_state=chain_state,
-            token_network_address=token_network_address,
-            from_address=from_address,
-            to_address=to_address,
-            amount=amount,
-            previous_address=previous_address,
+    token_network = views.get_token_network_by_address(chain_state, token_network_address)
+    assert token_network, "The token network must be validated and exist."
+
+    try:
+        # networkx returns a generator, consume the result since it will be
+        # iterated over multiple times.
+        all_neighbors = list(
```

Comment:
Any particular reason for using this over `views.all_neighbour_nodes()`?

Reply:
I didnt want to change the code, just to add the validation